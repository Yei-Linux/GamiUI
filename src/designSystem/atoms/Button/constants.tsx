export const options = {
  type: [
    "primary",
    "secondary",
    "tertiary",
    "cuaternary",
    "text",
    "link",
    "success",
    "warning",
    "danger",
    "info",
  ] as const,
};
