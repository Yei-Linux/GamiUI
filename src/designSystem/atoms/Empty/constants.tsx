export const iconsEmpty = {
  balloon: 'drawkit__empty',
  sad: 'sad__multicolor',
}

export const options = {
  type: ['balloon', 'sad'] as const,
}
