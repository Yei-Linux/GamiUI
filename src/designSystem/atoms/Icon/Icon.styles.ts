import styled from "@emotion/styled";
import { twinStyles } from "../../../styles/utilities/twinStyles";

export const IconWrapper = styled.i<{
  border?: string;
  shadow?: string;
}>`
  border: none;
  outline: none;
  cursor: pointer;
  appearance: none;
  display: flex;
  justify-content: center;
  align-items: center;

  ${(props: any) => twinStyles(props)};
`;
