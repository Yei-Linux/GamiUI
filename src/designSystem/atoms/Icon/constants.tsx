import React from "react";

export const IconsPack: any = {
  facebook: {
    viewBox: "0 0 20 20",
    svg: (
      <>
        <path d="M8.46 18H11.39V10.7H13.84L14.21 7.86H11.39V6.04C11.39 5.22 11.62 4.66 12.8 4.66H14.31V2.11C14.05 2.08 13.16 2 12.12 2C9.94 2 8.46 3.33 8.46 5.76V7.86H6V10.7H8.46V18Z" />
      </>
    ),
  },
  youtube: {
    viewBox: "0 0 20 20",
    svg: (
      <>
        <path d="M18.3848 5.7832C18.2851 5.41218 18.0898 5.07384 17.8184 4.80202C17.5469 4.53021 17.2088 4.33446 16.8379 4.23438C15.4727 3.86719 10 3.86719 10 3.86719C10 3.86719 4.52734 3.86719 3.16211 4.23242C2.79106 4.33219 2.45278 4.52782 2.18126 4.79969C1.90974 5.07155 1.71453 5.41007 1.61523 5.78125C1.25 7.14844 1.25 10 1.25 10C1.25 10 1.25 12.8516 1.61523 14.2168C1.81641 14.9707 2.41016 15.5645 3.16211 15.7656C4.52734 16.1328 10 16.1328 10 16.1328C10 16.1328 15.4727 16.1328 16.8379 15.7656C17.5918 15.5645 18.1836 14.9707 18.3848 14.2168C18.75 12.8516 18.75 10 18.75 10C18.75 10 18.75 7.14844 18.3848 5.7832ZM8.26172 12.6172V7.38281L12.793 9.98047L8.26172 12.6172Z" />
      </>
    ),
  },
  twitter: {
    viewBox: "0 0 20 20",
    svg: (
      <>
        <path d="M18.125 4.9668C17.5273 5.22462 16.877 5.41016 16.207 5.48243C16.9026 5.06922 17.4234 4.41641 17.6719 3.64649C17.0192 4.03475 16.3042 4.30701 15.5586 4.45118C15.247 4.11803 14.8701 3.85262 14.4514 3.67151C14.0327 3.49039 13.5812 3.39744 13.125 3.39845C11.2793 3.39845 9.79492 4.89454 9.79492 6.73048C9.79492 6.98829 9.82617 7.2461 9.87695 7.49415C7.11328 7.34962 4.64844 6.0293 3.00977 4.00782C2.71118 4.51781 2.55471 5.0985 2.55664 5.68946C2.55664 6.84571 3.14453 7.86524 4.04102 8.46485C3.5127 8.44405 2.99676 8.29883 2.53516 8.04102V8.08204C2.53516 9.70118 3.67969 11.043 5.20508 11.3516C4.91867 11.426 4.62404 11.464 4.32812 11.4649C4.11133 11.4649 3.90625 11.4434 3.69922 11.4141C4.12109 12.7344 5.34961 13.6934 6.8125 13.7246C5.66797 14.6211 4.23438 15.1484 2.67773 15.1484C2.39844 15.1484 2.14062 15.1387 1.87305 15.1074C3.34961 16.0547 5.10156 16.6016 6.98828 16.6016C13.1133 16.6016 16.4648 11.5274 16.4648 7.12305C16.4648 6.97852 16.4648 6.83399 16.4551 6.68946C17.1035 6.21485 17.6719 5.62696 18.125 4.9668Z" />
      </>
    ),
  },
  instagram: {
    viewBox: "0 0 20 20",
    svg: (
      <>
        <path d="M10.0001 7.39609C8.5665 7.39609 7.39658 8.56601 7.39658 9.9996C7.39658 11.4332 8.5665 12.6031 10.0001 12.6031C11.4337 12.6031 12.6036 11.4332 12.6036 9.9996C12.6036 8.56601 11.4337 7.39609 10.0001 7.39609ZM17.8087 9.9996C17.8087 8.92148 17.8185 7.85312 17.7579 6.77695C17.6974 5.52695 17.4122 4.41757 16.4981 3.50351C15.5821 2.58749 14.4747 2.30429 13.2247 2.24374C12.1466 2.1832 11.0782 2.19296 10.002 2.19296C8.92392 2.19296 7.85556 2.1832 6.77939 2.24374C5.52939 2.30429 4.42001 2.58945 3.50595 3.50351C2.58994 4.41953 2.30673 5.52695 2.24619 6.77695C2.18564 7.85507 2.19541 8.92343 2.19541 9.9996C2.19541 11.0758 2.18564 12.1461 2.24619 13.2223C2.30673 14.4723 2.59189 15.5816 3.50595 16.4957C4.42197 17.4117 5.52939 17.6949 6.77939 17.7555C7.85751 17.816 8.92587 17.8062 10.002 17.8062C11.0802 17.8062 12.1485 17.816 13.2247 17.7555C14.4747 17.6949 15.5841 17.4098 16.4981 16.4957C17.4142 15.5797 17.6974 14.4723 17.7579 13.2223C17.8204 12.1461 17.8087 11.0777 17.8087 9.9996ZM10.0001 14.0055C7.7833 14.0055 5.99423 12.2164 5.99423 9.9996C5.99423 7.78281 7.7833 5.99374 10.0001 5.99374C12.2169 5.99374 14.006 7.78281 14.006 9.9996C14.006 12.2164 12.2169 14.0055 10.0001 14.0055ZM14.17 6.76523C13.6524 6.76523 13.2345 6.34726 13.2345 5.82968C13.2345 5.3121 13.6524 4.89414 14.17 4.89414C14.6876 4.89414 15.1056 5.3121 15.1056 5.82968C15.1057 5.95258 15.0816 6.07431 15.0347 6.18788C14.9877 6.30146 14.9188 6.40465 14.8319 6.49156C14.745 6.57846 14.6418 6.64737 14.5282 6.69433C14.4146 6.74129 14.2929 6.76538 14.17 6.76523Z" />
      </>
    ),
  },
  wise: {
    viewBox: "0 0 512 512",
    svg: (
      <>
        <g>
          <g>
            <path
              d="m241 168.904v31.867c0 13.545 5.782 25.76 15 34.324l10-42.095-10-24.096z"
              fill="#b56a37"
            />
            <path
              d="m287.867 217.638c-9.301 0-16.867-7.566-16.867-16.867v-31.867h-15v66.191c8.369 7.776 19.569 12.543 31.867 12.543h37.32v-30z"
              fill="#8f532b"
            />
          </g>
          <g>
            <path
              d="m421.851 293.752h-86.06v-30h86.06c9.301 0 16.867-7.566 16.867-16.867v-106.189h30v106.189c0 25.843-21.025 46.867-46.867 46.867z"
              fill="#8f532b"
            />
          </g>
          <g>
            <path
              d="m177.621 278.482h-102.693c-25.843 0-46.867-21.024-46.867-46.867v-64.782h30v64.782c0 9.301 7.566 16.867 16.867 16.867h102.693z"
              fill="#b56a37"
            />
          </g>
          <path
            d="m175.935 212.807c-5.976-5.976-9.334-14.082-9.334-22.533v-42.615h-63.734v42.614c0 25.355 10.072 49.672 28.001 67.601l93.265 93.265v114.877h31.867l20-52.516-20-120.628z"
            fill="#c98353"
          />
          <path
            d="m324.521 144.966v66.185c0 8.452-3.357 16.557-9.334 22.534l-59.187 59.187v173.144h31.867v-114.877l72.387-72.387c17.929-17.929 28.001-42.245 28.001-67.601v-66.185z"
            fill="#b56a37"
          />
          <g>
            <path
              d="m256 31.867c0-17.6-14.267-31.867-31.867-31.867h-160.399c-35.199 0-63.734 28.535-63.734 63.734v22.387h77.178c-35.2 0-63.734 28.535-63.734 63.734v26.978h242.556l10-73.833-10-63.203z"
              fill="#9ce77a"
            />
            <path
              d="m512 82.287c0-35.2-28.535-74.357-63.734-74.357h-160.399c-17.6 0-31.867 14.268-31.867 31.867v137.036h53.521l15-26.137h151.781v-22.406c0-17.6-14.267-31.867-31.867-31.867h67.565z"
              fill="#66be61"
            />
          </g>
          <g>
            <g>
              <path
                d="m339.521 176.833h-30v-31.867c0-9.301-7.566-16.867-16.867-16.867h-22.845v-30h22.845c25.843 0 46.867 21.024 46.867 46.867z"
                fill="#009859"
              />
            </g>
          </g>
          <path
            d="m237.392 417.806c-15.76-6.693-32.706-10.143-49.829-10.143h-187.563v104.337h256l10-36.855-10-49.436z"
            fill="#ffe48d"
          />
          <path
            d="m512 448.266h-176.941c-17.122 0-34.069-3.45-49.829-10.143l-29.23-12.414v86.291h256z"
            fill="#f9cc74"
          />
        </g>
      </>
    ),
  },
  wizard: {
    viewBox: "0 0 512 512",
    svg: (
      <>
        <path
          d="m512 21.213-21.213-21.213-14.393 14.394-14.394-14.394-21.213 21.213 14.394 14.393-14.394 14.394 21.213 21.213 14.394-14.394 14.393 14.394 21.213-21.213-14.394-14.394z"
          fill="#ffdb2f"
        />
        <path
          d="m512 21.213-10.606-10.607-14.394 14.394-21.213 21.213-14.393 14.393 10.606 10.607 14.393-14.393 14.394 14.393 21.213-21.213-14.393-14.394z"
          fill="#eebf00"
        />
        <path
          d="m512 241.606-21.213-21.212-14.393 14.393-14.394-14.393-21.213 21.212 14.394 14.394-14.394 14.394 21.213 21.212 14.394-14.393 14.393 14.393 21.213-21.212-14.394-14.394z"
          fill="#eebf00"
        />
        <path
          d="m291.607 21.213-21.212-21.213-14.394 14.394-14.394-14.394-21.212 21.213 14.393 14.393-14.393 14.394 21.212 21.213 14.394-14.394 14.394 14.394 21.212-21.213-14.393-14.394z"
          fill="#ffda44"
        />
        <path
          d="m125.254 432.587-40.46-5.381-5.381-40.46 217.041-217.042 40.339 5.503 5.503 40.338z"
          fill="#009986"
        />
        <path
          d="m125.254 432.587-40.46-5.381 251.999-251.999 5.503 40.338z"
          fill="#006468"
        />
        <path
          d="m326.581 96.901h64.829v112.206h-64.829z"
          fill="#dce6eb"
          transform="matrix(.707 .707 -.707 .707 213.338 -209.034)"
        />
        <path
          d="m30.177 393.305h64.829v112.206h-64.829z"
          fill="#dce6eb"
          transform="matrix(.707 .707 -.707 .707 336.112 87.37)"
        />
        <g fill="#cde1e6">
          <path
            d="m354.249 108.361h32.414v112.206h-32.414z"
            transform="matrix(.707 .707 -.707 .707 224.798 -213.781)"
          />
          <path
            d="m57.844 404.766h32.414v112.206h-32.414z"
            transform="matrix(.707 .707 -.707 .707 347.573 82.623)"
          />
        </g>
      </>
    ),
  },
  killer: {
    viewBox: "0 0 512 512",
    svg: (
      <>
        <g>
          <g>
            <path
              d="m226.777 411.728c2.772 13.876 7.824 26.841 14.709 38.542l10-23.27-10-18.21z"
              fill="#3c4a52"
            />
            <path
              d="m444.053 372.715v14.437c0 52.299-42.549 94.848-94.849 94.848-45.078 0-84.193-32.025-93.009-76.149l-14.709 2.938v41.481c21.945 37.294 62.557 61.73 107.719 61.73 68.841 0 124.848-56.007 124.848-124.848v-14.437z"
              fill="#2d2929"
            />
          </g>
          <g>
            <path d="m189.882 64.445h30v58.463h-30z" fill="#3c4a52" />
          </g>
          <g>
            <path d="m263.092 64.445h30v58.463h-30z" fill="#2d2929" />
          </g>
          <g>
            <path
              d="m388.489 195.446h-83.793v-30h83.793c8.786 0 15.934-7.148 15.934-15.934v-30.622h30v30.623c0 25.327-20.605 45.933-45.934 45.933z"
              fill="#2d2929"
            />
          </g>
          <path
            d="m401 41.245-30.934 30.934v7.355c0 27.259 22.098 49.357 49.357 49.357 27.259 0 49.357-22.098 49.357-49.357v-17.667c0-34.168-27.699-61.867-61.867-61.867h-5.913z"
            fill="#3c4a52"
          />
          <g>
            <path
              d="m178.277 195.446h-83.793c-25.328 0-45.934-20.606-45.934-45.934v-30.622h30v30.623c0 8.786 7.147 15.934 15.934 15.934h83.794v29.999z"
              fill="#3c4a52"
            />
          </g>
          <path
            d="m81.973 41.245 30.934 30.934v7.355c0 27.259-22.098 49.357-49.357 49.357-27.259 0-49.357-22.098-49.357-49.357v-17.667c0-34.168 27.699-61.867 61.867-61.867h5.913z"
            fill="#566975"
          />
          <g>
            <g>
              <path
                d="m451.873 387.715c-25.328 0-45.934-20.605-45.934-45.934 0-25.328 20.605-45.934 45.934-45.934h15.934v-21.57h30v67.503c0 25.329-20.606 45.935-45.934 45.935z"
                fill="#3c4a52"
              />
            </g>
          </g>
          <g>
            <path
              d="m100.366 284.917-21.213-21.212 35.392-35.392h68.277v30h-55.851z"
              fill="#3c4a52"
            />
          </g>
          <g>
            <path
              d="m382.866 284.917-26.603-26.604h-55.852v-30h68.278l35.39 35.392z"
              fill="#2d2929"
            />
          </g>
          <g>
            <path
              d="m120.989 346.785-21.212-21.213 35.39-35.391h56.997v30h-44.571z"
              fill="#3c4a52"
            />
          </g>
          <g>
            <path
              d="m362.244 346.785-26.604-26.604h-44.831v-30h57.257l35.391 35.391z"
              fill="#2d2929"
            />
          </g>
          <g>
            <path
              d="m141.611 408.652-21.212-21.213 35.391-35.391h46.054v30h-33.628z"
              fill="#3c4a52"
            />
          </g>
          <g>
            <path
              d="m341.622 408.652-26.604-26.604h-33.888v-30h46.314l35.391 35.391z"
              fill="#2d2929"
            />
          </g>
          <g>
            <path
              d="m204.882 113.908c-20.216 0-36.605 16.388-36.605 36.605l6.157 52.473c1.559 13.286 8.182 18.623 12.842 24.224l-7.581 20.612 1.149 9.789 12.638 22.486-6.907 26.359 7.708 65.692c2.809 23.941 23.097 41.988 47.203 41.988l20-143.135-20-120.488c.001-20.217-16.388-36.605-36.604-36.605z"
              fill="#566975"
            />
            <path
              d="m278.091 113.908c-20.216 0-36.605 16.388-36.605 36.605v263.623c24.106 0 44.393-18.046 47.203-41.988l7.708-65.693-7.473-21.537 13.204-27.307 1.149-9.789-7.689-19.693c7.745-6.855 11.129-9.625 12.95-25.141l6.157-52.475c.001-20.217-16.387-36.605-36.604-36.605z"
              fill="#3c4a52"
            />
          </g>
          <g>
            <path
              d="m226.486 194.87c0 14.701-11.96 26.66-26.66 26.66-11.871 0-21.95-7.801-25.392-18.544l5.261 44.836c6.26 2.388 13.042 3.708 20.131 3.708 16.461 0 31.299-7.061 41.66-18.307l10-10.903-10-27.45z"
              fill="#3c4a52"
            />
            <path
              d="m283.147 251.53c7.088 0 13.871-1.32 20.131-3.708l5.261-44.834c-3.442 10.742-13.522 18.542-25.392 18.542-14.7 0-26.66-11.959-26.66-26.66h-15v38.353c10.361 11.246 25.199 18.307 41.66 18.307z"
              fill="#2d2929"
            />
            <path
              d="m180.844 257.611 5.732 48.844 47.211 48.409h7.7l10-17.376-10-17.697-.001.001z"
              fill="#3c4a52"
            />
            <path
              d="m249.187 354.864 47.211-48.409 5.731-48.844-60.642 62.18v35.073z"
              fill="#2d2929"
            />
          </g>
        </g>
      </>
    ),
  },
  closeeye: {
    viewBox: "0 0 512.52 512.52",
    svg: (
      <>
        <g>
          <g>
            <path
              d="m512.52 187.167-3.34 6.69c-.71 1.45-18.07 35.74-57.99 69.95-23.36 20.02-49.72 35.97-78.35 47.4-35.65 14.24-74.87 21.46-116.58 21.46s-80.93-7.22-116.58-21.46c-28.63-11.43-54.99-27.38-78.35-47.4-39.92-34.21-57.28-68.5-57.99-69.95l-3.34-6.69 3.34-6.69c.71-1.44 18.07-35.74 57.99-69.95 23.36-20.01 49.72-35.96 78.35-47.4 35.65-14.24 74.87-21.46 116.58-21.46s80.93 7.22 116.58 21.46c28.63 11.44 54.99 27.39 78.35 47.4 39.92 34.21 57.28 68.51 57.99 69.95z"
              fill="#fbf3f3"
            />
            <path
              d="m512.52 187.167-3.34 6.69c-.71 1.45-18.07 35.74-57.99 69.95-23.36 20.02-49.72 35.97-78.35 47.4-35.65 14.24-74.87 21.46-116.58 21.46v-291c41.71 0 80.93 7.22 116.58 21.46 28.63 11.44 54.99 27.39 78.35 47.4 39.92 34.21 57.28 68.51 57.99 69.95z"
              fill="#e8e3f0"
            />
            <ellipse
              cx="256.26"
              cy="187.167"
              fill="#adcfff"
              rx="114.12"
              ry="114.12"
              transform="matrix(.707 -.707 .707 .707 -57.291 236.023)"
            />
            <path
              d="m370.38 187.167c0 63.03-51.09 114.12-114.12 114.12v-228.24c63.03 0 114.12 51.101 114.12 114.12z"
              fill="#6aa8ff"
            />
            <ellipse
              cx="256.26"
              cy="189.437"
              fill="#363671"
              rx="59.42"
              ry="59.42"
              transform="matrix(.257 -.967 .967 .257 7.395 388.49)"
            />
            <path
              d="m315.68 189.437c0 32.82-26.6 59.42-59.42 59.42v-118.84c32.82 0 59.42 26.6 59.42 59.42z"
              fill="#16164d"
            />
            <circle cx="298.311" cy="151.391" fill="#fbf3f3" r="28.877" />
          </g>
          <ellipse
            cx="412.11"
            cy="370.703"
            fill="#ff7c48"
            rx="100.15"
            ry="100.15"
            transform="matrix(.707 -.707 .707 .707 -141.422 399.982)"
          />
          <path
            d="m512.26 370.703c0 55.22-44.93 100.15-100.15 100.15v-200.3c55.22 0 100.15 44.92 100.15 100.15z"
            fill="#ff3939"
          />
          <path
            d="m412.11 355.707 10 14.996-10 15.004h-41.48v-30z"
            fill="#fbf3f3"
          />
          <path d="m412.11 355.707h41.48v30h-41.48z" fill="#e8e3f0" />
        </g>
      </>
    ),
  },
  openeye: {
    viewBox: "0 0 512.52 512.52",
    svg: (
      <>
        <g>
          <g>
            <path
              d="m512.52 187.167-3.34 6.69c-.71 1.45-18.07 35.74-57.99 69.95-23.36 20.02-49.72 35.97-78.35 47.4-35.65 14.24-74.87 21.46-116.58 21.46s-80.93-7.22-116.58-21.46c-28.63-11.43-54.99-27.38-78.35-47.4-39.92-34.21-57.28-68.5-57.99-69.95l-3.34-6.69 3.34-6.69c.71-1.44 18.07-35.74 57.99-69.95 23.36-20.01 49.72-35.96 78.35-47.4 35.65-14.24 74.87-21.46 116.58-21.46s80.93 7.22 116.58 21.46c28.63 11.44 54.99 27.39 78.35 47.4 39.92 34.21 57.28 68.51 57.99 69.95z"
              fill="#fbf3f3"
            />
            <path
              d="m512.52 187.167-3.34 6.69c-.71 1.45-18.07 35.74-57.99 69.95-23.36 20.02-49.72 35.97-78.35 47.4-35.65 14.24-74.87 21.46-116.58 21.46v-291c41.71 0 80.93 7.22 116.58 21.46 28.63 11.44 54.99 27.39 78.35 47.4 39.92 34.21 57.28 68.51 57.99 69.95z"
              fill="#e8e3f0"
            />
            <ellipse
              cx="256.26"
              cy="187.167"
              fill="#adcfff"
              rx="114.12"
              ry="114.12"
              transform="matrix(.707 -.707 .707 .707 -57.291 236.023)"
            />
            <path
              d="m370.38 187.167c0 63.03-51.09 114.12-114.12 114.12v-228.24c63.03 0 114.12 51.101 114.12 114.12z"
              fill="#6aa8ff"
            />
            <ellipse
              cx="256.26"
              cy="189.437"
              fill="#363671"
              rx="59.42"
              ry="59.42"
              transform="matrix(.257 -.967 .967 .257 7.395 388.49)"
            />
            <path
              d="m315.68 189.437c0 32.82-26.6 59.42-59.42 59.42v-118.84c32.82 0 59.42 26.6 59.42 59.42z"
              fill="#16164d"
            />
            <circle cx="298.311" cy="151.391" fill="#fbf3f3" r="28.877" />
          </g>
          <ellipse
            cx="412.11"
            cy="370.703"
            fill="#9bdc05"
            rx="100.15"
            ry="100.15"
            transform="matrix(.707 -.707 .707 .707 -141.422 399.982)"
          />
          <path
            d="m512.26 370.703c0 55.22-44.93 100.15-100.15 100.15v-200.3c55.22 0 100.15 44.92 100.15 100.15z"
            fill="#00cb76"
          />
          <path
            d="m412.11 329.223h-15v26.48h-26.48v30h26.48v26.48h15l8.324-26.48v-30z"
            fill="#fbf3f3"
          />
          <path
            d="m453.59 355.703v30h-26.48v26.48h-15v-82.96h15v26.48z"
            fill="#e8e3f0"
          />
        </g>
      </>
    ),
  },
  padlock: {
    viewBox: "0 0 512 512",
    svg: (
      <>
        <path
          style={{ fill: "#455A64" }}
          d="M256,0c-82.436,0.094-149.239,66.898-149.333,149.333v96c0,5.891,4.776,10.667,10.667,10.667H160
	c5.891,0,10.667-4.776,10.667-10.667v-96C170.667,102.205,208.872,64,256,64s85.333,38.205,85.333,85.333v96
	c0,5.891,4.776,10.667,10.667,10.667h42.667c5.891,0,10.667-4.776,10.667-10.667v-96C405.239,66.898,338.436,0.094,256,0z"
        />
        <path
          style={{ fill: "#FFC107" }}
          d="M394.667,234.667H117.333c-17.673,0-32,14.327-32,32v192c0,29.455,23.878,53.333,53.333,53.333
	h234.667c29.455,0,53.333-23.878,53.333-53.333v-192C426.667,248.994,412.34,234.667,394.667,234.667z"
        />
        <path
          style={{ fill: "#455A64" }}
          d="M284.821,372.693c8.858-8.014,13.892-19.415,13.845-31.36c0-23.564-19.103-42.667-42.667-42.667
	s-42.667,19.103-42.667,42.667c-0.046,11.945,4.987,23.346,13.845,31.36l-13.312,39.936c-1.862,5.589,1.16,11.629,6.749,13.491
	c1.091,0.363,2.234,0.548,3.384,0.547h64c5.891,0.007,10.673-4.762,10.68-10.653c0.001-1.15-0.183-2.293-0.547-3.384
	L284.821,372.693z"
        />
        <g></g>
        <g></g>
        <g></g>
        <g></g>
        <g></g>
        <g></g>
        <g></g>
        <g></g>
        <g></g>
        <g></g>
        <g></g>
        <g></g>
        <g></g>
        <g></g>
        <g></g>
      </>
    ),
  },
  padlockBlack: {
    viewBox: "0 0 24 24",
    svg: (
      <>
        <path d="m18.75 9h-.75v-3c0-3.309-2.691-6-6-6s-6 2.691-6 6v3h-.75c-1.24 0-2.25 1.009-2.25 2.25v10.5c0 1.241 1.01 2.25 2.25 2.25h13.5c1.24 0 2.25-1.009 2.25-2.25v-10.5c0-1.241-1.01-2.25-2.25-2.25zm-10.75-3c0-2.206 1.794-4 4-4s4 1.794 4 4v3h-8zm5 10.722v2.278c0 .552-.447 1-1 1s-1-.448-1-1v-2.278c-.595-.347-1-.985-1-1.722 0-1.103.897-2 2-2s2 .897 2 2c0 .737-.405 1.375-1 1.722z" />
      </>
    ),
  },
  upset: {
    viewBox: "0 0 512 512",
    svg: (
      <>
        <g>
          <g>
            <g>
              <g id="XMLID_3037_">
                <g id="XMLID_3038_">
                  <g id="XMLID_3039_">
                    <g id="XMLID_3040_">
                      <g id="XMLID_3041_">
                        <g id="XMLID_3042_">
                          <g id="XMLID_3043_">
                            <g id="XMLID_3045_">
                              <g id="XMLID_3046_">
                                <g id="XMLID_3047_">
                                  <g id="XMLID_3048_">
                                    <g id="XMLID_3049_">
                                      <g id="XMLID_3050_">
                                        <g id="XMLID_3051_">
                                          <g id="XMLID_3072_">
                                            <circle
                                              cx="256"
                                              cy="256"
                                              fill="#4f4359"
                                              r="256"
                                            />
                                          </g>
                                        </g>
                                      </g>
                                    </g>
                                  </g>
                                </g>
                              </g>
                            </g>
                          </g>
                        </g>
                      </g>
                    </g>
                  </g>
                </g>
              </g>
            </g>
          </g>
          <path
            d="m511.98 257.604-125.738-125.737-254.376 254.376 125.737 125.737c140.11-.862 253.515-114.266 254.377-254.376z"
            fill="#372a44"
          />
          <g>
            <g>
              <g>
                <g id="XMLID_2899_">
                  <g id="XMLID_2900_">
                    <g id="XMLID_2916_">
                      <g id="XMLID_2917_">
                        <g id="XMLID_2947_">
                          <g id="XMLID_2949_">
                            <g id="XMLID_2950_">
                              <g id="XMLID_2952_">
                                <g id="XMLID_2953_">
                                  <g id="XMLID_2954_">
                                    <g id="XMLID_2955_">
                                      <g id="XMLID_2956_">
                                        <g id="XMLID_2957_">
                                          <g id="XMLID_2988_">
                                            <g id="XMLID_3006_">
                                              <circle
                                                cx="255.999"
                                                cy="256"
                                                fill="#fed940"
                                                r="179.923"
                                              />
                                            </g>
                                          </g>
                                        </g>
                                      </g>
                                    </g>
                                  </g>
                                </g>
                              </g>
                            </g>
                          </g>
                        </g>
                      </g>
                    </g>
                  </g>
                </g>
              </g>
            </g>
            <g>
              <g>
                <g id="XMLID_2832_">
                  <g id="XMLID_2833_">
                    <g id="XMLID_2834_">
                      <g id="XMLID_2835_">
                        <g id="XMLID_2836_">
                          <g id="XMLID_2837_">
                            <g id="XMLID_2853_">
                              <g id="XMLID_2869_">
                                <g id="XMLID_2870_">
                                  <g id="XMLID_2871_">
                                    <g id="XMLID_2872_">
                                      <g id="XMLID_2888_">
                                        <g id="XMLID_2889_">
                                          <g id="XMLID_2890_">
                                            <g>
                                              <path
                                                d="m435.923 256c0 99.367-80.557 179.924-179.924 179.924v-359.848c99.368 0 179.924 80.557 179.924 179.924z"
                                                fill="#ffc024"
                                              />
                                            </g>
                                          </g>
                                        </g>
                                      </g>
                                    </g>
                                  </g>
                                </g>
                              </g>
                            </g>
                          </g>
                        </g>
                      </g>
                    </g>
                  </g>
                </g>
              </g>
            </g>
            <path
              d="m344.61 328.63-30.41 17.45c-12.64-18.67-34.01-30.95-58.2-30.95s-45.56 12.28-58.2 30.95l-30.41-17.45c18.74-29.15 51.46-48.5 88.61-48.5s69.87 19.35 88.61 48.5z"
              fill="#7f6866"
            />
            <path
              d="m344.61 328.63-30.41 17.45c-12.64-18.67-34.01-30.95-58.2-30.95v-35c37.15 0 69.87 19.35 88.61 48.5z"
              fill="#634c4c"
            />
            <g>
              <g>
                <path
                  d="m165.85 250.121c-10.439 0-20.867-1.956-30.646-5.843l11.08-27.879c23.017 9.145 48.938 1.314 63.036-19.054l24.668 17.074c-10.623 15.347-26.314 26.772-44.18 32.174-7.794 2.356-15.88 3.528-23.958 3.528z"
                  fill="#7f6866"
                />
              </g>
              <g>
                <path
                  d="m346.148 250.121c-8.079 0-16.163-1.172-23.958-3.527-17.865-5.401-33.556-16.827-44.179-32.174l24.668-17.074c14.097 20.365 40.018 28.202 63.035 19.054l11.08 27.879c-9.777 3.885-20.207 5.842-30.646 5.842z"
                  fill="#634c4c"
                />
              </g>
            </g>
          </g>
        </g>
      </>
    ),
  },
  close: {
    viewBox: "0 0 329.26933 329",
    svg: (
      <>
        <path d="m194.800781 164.769531 128.210938-128.214843c8.34375-8.339844 8.34375-21.824219 0-30.164063-8.339844-8.339844-21.824219-8.339844-30.164063 0l-128.214844 128.214844-128.210937-128.214844c-8.34375-8.339844-21.824219-8.339844-30.164063 0-8.34375 8.339844-8.34375 21.824219 0 30.164063l128.210938 128.214843-128.210938 128.214844c-8.34375 8.339844-8.34375 21.824219 0 30.164063 4.15625 4.160156 9.621094 6.25 15.082032 6.25 5.460937 0 10.921875-2.089844 15.082031-6.25l128.210937-128.214844 128.214844 128.214844c4.160156 4.160156 9.621094 6.25 15.082032 6.25 5.460937 0 10.921874-2.089844 15.082031-6.25 8.34375-8.339844 8.34375-21.824219 0-30.164063zm0 0" />
      </>
    ),
  },
  arrowright: {
    viewBox: "0 0 24 24",
    svg: (
      <>
        <path d="M10 6L8.59 7.41L13.17 12l-4.58 4.59L10 18l6-6l-6-6z" fill="#626262"/>
      </>
    ),
  },
};

export const options = {
  names: Object.keys(IconsPack),
};
