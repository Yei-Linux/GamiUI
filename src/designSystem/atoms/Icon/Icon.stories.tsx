import { Meta } from "@storybook/react";
import { getTemplate, stylesControl } from "../../../core/helpers/storybook";

import Icon from ".";
import { options as optionsIcon } from "./constants";

const Template = getTemplate(Icon);

export default {
  title: "Atoms/Icon",
  component: Icon,
  argTypes: {
    children: { control: "text" },
    name: { control: { type: "select", options: optionsIcon.names } },
    ...stylesControl,
  },
} as Meta;

export const BasicIcon = Template.bind({});
BasicIcon.args = {
  fill: "red",
  size: "25px",
  name: "facebook",
};
