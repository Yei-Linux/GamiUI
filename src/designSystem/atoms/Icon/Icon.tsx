import React, { useEffect, useState } from "react";
import { IGeneralProps } from "../../../core/domain/interfaces/IGeneralProps";
import { IconNames } from "../../../core/domain/types";
import { IconsPack } from "./constants";
import { IconWrapper } from "./Icon.styles";

interface IIcon extends IGeneralProps {
  /**
   * Fill Icon
   */
  fill: string;
  /**
   * Name
   */
  name: IconNames;
  /**
   * Size icon
   */
  size: string;
}

const Svg = ({
  children,
  ...props
}: {
  children: React.ReactNode;
  width: string;
  height: string;
  viewBox: string;
  fill: string;
}) => {
  return (
    <svg xmlns="http://www.w3.org/2000/svg" {...props}>
      {children}
    </svg>
  );
};

const Icon = ({ fill, name, size, ...args }: IIcon) => {
  const [icon, setIcon] = useState(IconsPack[name]);

  useEffect(() => {
    setIcon(IconsPack[name]);
  }, [name]);

  return (
    <IconWrapper {...args}>
      <Svg fill={fill} width={size} height={size} viewBox={icon.viewBox}>
        {icon.svg}
      </Svg>
    </IconWrapper>
  );
};

Icon.defaultProps = {
  fill: "none",
  name: "facebook",
  size: "16px",
};

export default Icon;