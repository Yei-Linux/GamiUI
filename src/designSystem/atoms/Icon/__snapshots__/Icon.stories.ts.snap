// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Icon Basic Icon 1`] = `
<i
  className="css-6tw2qr-IconWrapper e1cr9qb30"
>
  <svg
    fill="red"
    height="25px"
    viewBox="0 0 20 20"
    width="25px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8.46 18H11.39V10.7H13.84L14.21 7.86H11.39V6.04C11.39 5.22 11.62 4.66 12.8 4.66H14.31V2.11C14.05 2.08 13.16 2 12.12 2C9.94 2 8.46 3.33 8.46 5.76V7.86H6V10.7H8.46V18Z"
    />
  </svg>
</i>
`;
